worlds/wargroove2/Client.py

```diff
@@ -0,0 +1,858 @@
+                        f.write(f"{item_count * self.commander_defense_boost_multiplier}")
+                    elif self.buff_item_ids["Groove Boost"] == network_item.item:
+                        f.write(f"{item_count * self.starting_groove_multiplier}")
+                    else:
+                        f.write(f"{item_count}")
+
+                print_filename = f"AP_{network_item.item}.item.print"
+                print_path = os.path.join(self.game_communication_path, print_filename)
+                if not os.path.isfile(print_path):
+                    open(print_path, 'w').close()
+                    with open(print_path, 'w') as f:
+                        f.write("Received " +
+                                self.item_names.lookup_in_slot(network_item.item) +
+                                " from " +
+                                self.player_names[network_item.player])
+            self.update_commander_data()
+            self.ui.update_ui()
+
+        if cmd in {"RoomUpdate"}:
+            if "checked_locations" in args:
+                for ss in self.checked_locations:
+                    filename = f"send{ss}"
+                    with open(os.path.join(self.game_communication_path, filename), 'w') as f:
+                        pass
+            self.ui.update_ui()
+
+        if cmd in {"Retrieved"}:
+            self.ui.update_levels()
+
+    def run_gui(self):
+        """Import kivy UI system and start running it as self.ui_task."""
+        from kvui import GameManager
+        from kivymd.uix.tab import MDTabsItem, MDTabsItemText
+        from kivy.uix.tabbedpanel import TabbedPanelItem
+        from kivy.lang import Builder
+        from kivy.uix.togglebutton import ToggleButton
+        from kivy.uix.boxlayout import BoxLayout
+        from kivy.uix.gridlayout import GridLayout
+        from kivy.uix.label import Label
+        import pkgutil
+
+        class TrackerLayout(BoxLayout):
+            pass
+
+        class LevelsLayout(BoxLayout):
+            pass
+
+        class CommanderSelect(BoxLayout):
+            pass
+
+        class CommanderButton(ToggleButton):
+            pass
+
+        class FactionBox(BoxLayout):
+            pass
+
+        class CommanderGroup(BoxLayout):
+            pass
+
+        class ItemTracker(BoxLayout):
+            pass
+
+        class LevelTracker(BoxLayout):
+            pass
+
+        class ItemLabel(Label):
+            pass
+
+        class Wargroove2Manager(GameManager):
+            logging_pairs = [
+                ("Client", "Archipelago"),
+                ("WG2", "WG2 Console"),
+            ]
+            base_title = "Archipelago Wargroove 2 Client"
+            ctx: Wargroove2Context
+            unit_tracker: ItemTracker
+            level_tracker: LevelTracker
+            level_1_Layout: GridLayout(cols=1)
+            level_2_Layout: GridLayout(cols=1)
+            level_3_Layout: GridLayout(cols=1)
+            level_4_Layout: GridLayout(cols=1)
```

Comment:
```suggestion
            level_1_Layout: GridLayout
            level_2_Layout: GridLayout
            level_3_Layout: GridLayout
            level_4_Layout: GridLayout
```
I don't think there's a reason for the hints to be instantiated like this.